### PR TITLE
Application active

### PR DESCRIPTION
**Pull request by eguzki**
Disable application:

```
require 'pisoni'

ENV['THREESCALE_CORE_INTERNAL_API'] = nil
ThreeScale::Core.url = "http://myhost:myport"
ThreeScale::Core.username = "username_key"
ThreeScale::Core.password = "password_key"

application = ThreeScale::Core::Application.load(service_id, app_id)

application.deactivate
application.save
```

Activate application:

```
require 'pisoni'

ENV['THREESCALE_CORE_INTERNAL_API'] = nil
ThreeScale::Core.url = "http://myhost:myport"
ThreeScale::Core.username = "username_key"
ThreeScale::Core.password = "password_key"

application = ThreeScale::Core::Application.load(service_id, app_id)

application.activate
application.save
```

Activation status can be queried

```
require 'pisoni'

ENV['THREESCALE_CORE_INTERNAL_API'] = nil
ThreeScale::Core.url = "http://myhost:myport"
ThreeScale::Core.username = "username_key"
ThreeScale::Core.password = "password_key"

application = ThreeScale::Core::Application.load(service_id, app_id)

puts application.active?
```

### CODE REVIEW

File: spec/application_spec.rb [hunk 1]
```diff
@@ -152,6 +152,64 @@ module Core
...
         end
 
+        it 'save active application' do
+          @app.activate
+          @app.save
+          newapp = Application.load @app.service_id, @app.id
+          newapp.active?.must_equal true
+        end
+
+        it 'save disable application' do
```
Comment (unleashed) [inline spec/application_spec.rb]: inactive?

File: spec/application_spec.rb [hunk 2]
```diff
@@ -152,6 +152,64 @@ module Core
...
+        end
+
+        it 'should be active when the application is activated' do
+          app = Application.new(state: :disable)
+          app.active?.must_equal false
+          app.activate
+          app.active?.must_equal true
+        end
+
+        it 'should be disabled when the application is deactivated' do
```
Comment (unleashed) [inline spec/application_spec.rb]: `should be inactive` here as well?

**Review by unleashed — APPROVED** [review]
check the two minor comments about wording and LGTM

File: spec/application_spec.rb [hunk 3]
```diff
@@ -152,6 +152,64 @@ module Core
...
+          newapp = Application.load @app.service_id, @app.id
+          newapp.active?.must_equal true
+        end
+
+        it 'save inactive application' do
+          @app.deactivate
+          @app.save
+          newapp = Application.load @app.service_id, @app.id
+          newapp.active?.must_equal false
+        end
```
Comment (davidor) [inline spec/application_spec.rb:167]: Should we add a test checking that the app is active when we don't activate or deactivate it?

File: spec/application_spec.rb [hunk 4]
```diff
@@ -152,6 +152,64 @@ module Core
...
+          app.activate
+          app.active?.must_equal true
+        end
+
+        it 'should be inactive when the application is deactivated' do
+          app = Application.new(state: :active)
+          app.active?.must_equal true
+          app.deactivate
+          app.active?.must_equal false
+        end
```
Comment (davidor) [inline spec/application_spec.rb:225]: Same as above. Should we test that the app is active by default?

Comment (davidor) [reply spec/application_spec.rb:225]: I noticed that this is already tested in the first test of this block :+1:

File: spec/application_spec.rb (hunk 3 above)
Comment (eguzki) [reply spec/application_spec.rb:167]: added test to save application with no explicit state (when we don't activate or deactivate). On loading, it should be active

**Review by davidor — APPROVED** [review]

**Comment by unleashed** [comment]
FYI @eguzki `bors r+` is the same as `bors r=@eguzki` :)

**Pull request merged**